app/domain/usecases/create_account_test.go

```diff
@@ -0,0 +1,143 @@
+package usecase
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/jpgsaraceni/suricate-bank/app/domain/entities/account"
+)
+
+func TestCreate(t *testing.T) {
+	t.Parallel()
+
+	type args struct {
+		name   string
+		cpf    string
+		secret string
+	}
+
+	type testCase struct {
+		name       string
+		repository account.Repository
+		args       args
+		want       account.Account
+		err        error
+	}
+
+	var testUUID, _ = uuid.NewUUID()
+	var testAccountId = account.AccountId(testUUID)
+	var testTime = time.Now()
+
+	var mockCreateSuccess = func(account *account.Account) error {
+		account.Id = testAccountId
+		account.CreatedAt = testTime
+		account.Secret = "hashedpassphrase"
+		return nil
+	}
+
+	testCases := []testCase{
+		{
+			name: "successfully create account",
+			repository: account.MockRepository{
+				OnCreate: mockCreateSuccess,
+			},
+			args: args{
+				name:   "meee",
+				cpf:    "220.614.460-35",
+				secret: "reallygoodpassphrase",
+			},
+			want: account.Account{
+				Name:      "meee",
+				Cpf:       "22061446035",
+				Id:        testAccountId,
+				CreatedAt: testTime,
+				Secret:    "hashedpassphrase",
+			},
+		},
+		{
+			name: "fail to create account because password is too short",
+			repository: account.MockRepository{
+				OnCreate: mockCreateSuccess,
+			},
+			args: args{
+				name:   "meee",
+				cpf:    "220.614.460-35",
+				secret: "123",
+			},
+			want: account.Account{},
+			err:  errShortSecret,
+		},
```

Comment:
Como essa é uma validação que acontece antes de chamar o repository, passar `nil` na func do mock pode ser um jeito de verificar o repository não está sendo chamado.
```suggestion
		{
			name: "fail to create account because password is too short",
			repository: account.MockRepository{
				OnCreate: nil,
			},
			args: args{
				name:   "meee",
				cpf:    "220.614.460-35",
				secret: "123",
			},
			want: account.Account{},
			err:  errShortSecret,
		},
```